proposals/awslogs.md

```diff
@@ -0,0 +1,195 @@
+
+Work Flow:
+1.	User gives Task ID
+2.	Call Describe Tasks to get the TaskDef ARN (Skip this step if user provides Task Def)
+3.	Call Describe Task Definition to get the Container Definitions.
+4.	From Container Definitions, get the log configuration.
+5.	Create a list of log streams that correspond to the correct task for the log group.
+6.	Call FilterLogEvents for each log group to get the log events.
+7.	Print log events.
+
+
+## Compose Logs (Phase 2)
+Phase 2 will be implemented in the future when we have time, it is lower priority than Phase 1, and thus Phase 2 may not be implemented for some time. We welcome the contributions of any customer who wishes to help start implementation of Phase 2 sooner.
+
+### Configure Logs
+- Log configuration using the docker-compose file is already supported
+- Problem: Customer not required specify log stream prefix, however, we basically need log stream prefix to be specified because of how the ECS Agent sets the log stream name. If prefix is specified then it adds the container name and task ID to the log stream name (so we can use it to get the logs for each task). However, if a prefix is not specified, then the log stream name will for all intents and purposes be a random useless string (its an ID picked by the docker daemon on the instance, which from our point of view is meaningless).
+    - The log stream will be named like this (by the ECS Agent): `prefix-name/container-name/ecs-task-id`
+
+*Solution:* Existing ability to configure logs remains undisturbed, but add additional flag `--create-logs` that creates the necessary log group(s) in CloudWatch.
+- The log configuration from the docker compose file will be read
+- If user has not specified a log stream prefix, warn them that we are auto-setting it to a default value in their task definition.
+    - *Additionally*, even if `--create-logs` is not specified, but we detect that the there is no prefix configured in their docker compose file (but log group and awslogs driver is specified), prefix will still be auto-set, and the user will be warned about this. This technically will break backwards compatibility- however, this risk is acceptable. It is very unlikely that ECS CLI users would actually desire to have their log streams named without a prefix. If no prefix is given, the ECS Agent sets the log stream name to be the container ID which was randomly generated by docker. Understanding this random ID requires logging into the underlying instances and retrieving info from the Docker Daemon. For all intents and purposes, the container ID is meaningless from a customer standpoint.
+    - The ECS CLI is designed to simplify workflows and make it easier to understand ECS. Therefore, we should be opinionated and protect users from accidentally configuring there logs in a poor way. We can help protect users from the less useful, complicated, legacy behavior of ECS.
```

Comment:
> It is very unlikely that ECS CLI users would actually desire to have their log streams named without a prefix
Do we know if this is actually true? Seems like there's nothing that prevents a customer from naming a stream without a prefix, but not super familiar with CWlog configuration.

Reply:
I am also a teensy bit nervous about calling the flag `--create-logs` because there is both log groups and log streams, which are different ideas, so it might be a tad confusing.

Reply:
`--create-log-groups` instead?

Reply:
Also, regarding the prefixes, this statement is definitely true in my opinion: "It is very unlikely that ECS CLI users would actually desire to have their log streams named without a prefix". Because without a prefix the log stream gets a random ID for its name.

More importantly, we simply can't build a useful logging feature unless the prefix is specified.